lib/api/src/lib/stories.ts

```diff
@@ -144,103 +136,89 @@ export const transformStoriesRawToStoriesHash = (
   input: StoriesRaw,
   { provider }: { provider: Provider }
 ): StoriesHash => {
-  const anyKindMatchesOldHierarchySeparators = Object.values(input)
-    .filter(Boolean)
-    .some(({ kind }) => kind.match(/\.|\|/));
+  const values = Object.values(input).filter(Boolean);
+  const usesOldHierarchySeparator = values.some(({ kind }) => kind.match(/\.|\|/)); // dot or pipe
 
-  const storiesHashOutOfOrder = Object.values(input)
-    .filter(Boolean)
-    .reduce((acc, item) => {
-      const { kind, parameters } = item;
-      const { showRoots } = provider.getConfig();
+  const storiesHashOutOfOrder = values.reduce((acc, item) => {
+    const { kind, parameters } = item;
+    const { showRoots } = provider.getConfig();
 
-      const setShowRoots = typeof showRoots !== 'undefined';
-      if (anyKindMatchesOldHierarchySeparators && !setShowRoots) {
-        warnChangedDefaultHierarchySeparators();
-      }
+    const setShowRoots = typeof showRoots !== 'undefined';
+    if (usesOldHierarchySeparator && !setShowRoots) {
+      warnChangedDefaultHierarchySeparators();
+    }
 
-      let root = '';
-      let groups: string[];
-      const parts: string[] = kind.split('/');
-      // Default showRoots to true if they didn't set it.
-      if ((!setShowRoots || showRoots) && parts.length > 1) {
-        [root, ...groups] = parts;
-      } else {
-        groups = parts;
-      }
+    const groups = kind.split('/').map((part) => part.trim());
+    const root = (!setShowRoots || showRoots) && groups.length > 1 ? [groups.shift()] : [];
```

Comment:
Using `array.shift()` allows us to avoid `let` and greatly simplify the way we assign these two variables. `root` gets wrapped in an array so we can directly spread it below, avoiding the odd-looking `concat(root || [])` logic.